crates/transaction-emitter-lib/src/emitter/mod.rs

```diff
@@ -602,6 +601,37 @@ impl TxnEmitter {
     }
 }
 
+pub struct RestApiTransactionExecutor {
+    rest_client: RestClient,
+}
+
+#[async_trait]
+impl TransactionExecutor for RestApiTransactionExecutor {
+    async fn execute_transactions(&self, txns: &[SignedTransaction]) -> Result<()> {
+        join_all(txns.iter().map(|txn| async move {
+            let submit_result = RETRY_POLICY
+                .retry(move || self.rest_client.submit_bcs(txn))
+                .await;
+            if let Err(e) = submit_result {
+                warn!("Failed submitting transaction {:?} with {:?}", txn, e);
+            }
+        }))
+        .await;
+
+        // if submission timeouts, it might still get committed:
+        join_all(
+            txns.iter()
+                .map(|req| self.rest_client.wait_for_signed_transaction_bcs(req)),
```

Comment:
Do we need any retry here as well?

Reply:
no, it does retries internally. 

if it returns, it means transaction will never be committed 

(there is an optional argument how stale a full node can be, and we can increase that/set it to None if that becomes an issue, but that needs to have lag in minutes)